src/tests/payment/test_stripe_webhook.py

```diff
@@ -0,0 +1,430 @@
+            == "null"
+            else timezone.make_aware(
+                datetime.fromtimestamp(
+                    customer_subscription_updated_payload["data"]["object"]["trial_end"]
+                )
+            )
+        )
+        assert subscription.plan == bool(
+            customer_subscription_updated_payload["data"]["object"]["items"]["data"][0][
+                "plan"
+            ]["active"]
+        )
+
+    @mock.patch("stripe.Webhook.construct_event")
+    def test_invoice_paid(
+        self, mock_construct_event, invoice_paid_payload, client, user
+    ):
+        """
+        Test the `invoice.paid` event.
+        Verify that the `hosted_invoice_url` and `invoice_pdf` fields
+        of the subscription are correctly updated.
+        """
+
+        CustomerDetail.objects.create(
+            user=user,
+            stripe_customer_id=invoice_paid_payload["data"]["object"]["customer"],
+        )
+
+        Subscription.objects.create(
+            user=user, status="active", hosted_invoice_url="", invoice_pdf=""
+        )
+
+        sig_header = generate_stripe_signature(
+            invoice_paid_payload, STRIPE_WEBHOOK_SECRET
+        )
+        mock_construct_event.return_value = invoice_paid_payload
+
+        response = client.post(
+            self.url,
+            data=json.dumps(invoice_paid_payload),
+            content_type="application/json",
+            HTTP_STRIPE_SIGNATURE=sig_header,
+        )
+
+        assert response.status_code == 200
+
+        subscription = Subscription.objects.get(user=user)
+        assert (
+            subscription.hosted_invoice_url
+            == invoice_paid_payload["data"]["object"]["hosted_invoice_url"]
+        )
+        assert (
+            subscription.invoice_pdf
+            == invoice_paid_payload["data"]["object"]["invoice_pdf"]
+        )
+
+    @mock.patch("stripe.Webhook.construct_event")
+    def test_customer_subscription_deleted(
+        self, mock_construct_event, customer_subscription_deleted_payload, client, user
+    ):
+        """
+        Test the `customer.subscription.deleted` event.
+        Verify that the subscription status is correctly updated to `Canceled`.
+        """
+
+        CustomerDetail.objects.create(
+            user=user,
+            stripe_customer_id=customer_subscription_deleted_payload["data"]["object"][
+                "customer"
+            ],
+        )
+
+        Subscription.objects.create(user=user, status="active")
+
+        sig_header = generate_stripe_signature(
+            customer_subscription_deleted_payload, STRIPE_WEBHOOK_SECRET
+        )
+        mock_construct_event.return_value = customer_subscription_deleted_payload
+
+        response = client.post(
+            self.url,
+            data=json.dumps(customer_subscription_deleted_payload),
+            content_type="application/json",
+            HTTP_STRIPE_SIGNATURE=sig_header,
+        )
+
+        assert response.status_code == 200
```

Comment:
`status.HTTP_200_OK` here and all the other missing places

Reply:
I've updated the code throughout the file to use status.HTTP_200_OK consistently wherever applicable